tests/test_module_loader.py

```diff
@@ -127,3 +128,130 @@ def test_v1_module_depends_on_v1_and_v2_module(tmpdir: py.path.local, snippetcom
+    project: Project = snippetcompiler_clean.setup_for_snippet(
+        snippet=f"import {module_name}", install_v2_modules=[LocalPackagePath(module_dir, editable_install)]
+    )
+
+    assert module_name not in project.modules
+    project.load_module(module_name=module_name, install=False, allow_v1=allow_v1)
+    assert module_name in project.modules
+
+
+@pytest.mark.parametrize("install", [True, False])
+def test_load_module_v2_module_using_install(
+    snippetcompiler_clean,
+    local_module_package_index: str,
+    install: bool,
+):
+    """
+    Test whether the Project.load_module() method works correctly when a module is only available as a V2 module
+    and that module is not yet installed in the compiler venv.
+    """
+    module_name = "minimalv2module"
+    project: Project = snippetcompiler_clean.setup_for_snippet(
+        snippet=f"import {module_name}", python_package_source=local_module_package_index
+    )
+    assert module_name not in project.modules
+    assert module_name not in os.listdir(project.downloadpath)
+    if install:
+        project.load_module(module_name=module_name, install=install, allow_v1=True)
+    else:
+        with pytest.raises(ModuleNotFoundException, match=f"Could not find module {module_name}"):
+            project.load_module(module_name=module_name, install=install, allow_v1=True)
+    assert (module_name in project.modules) == install
+    assert module_name not in os.listdir(project.downloadpath)
+
+
+@pytest.mark.parametrize("allow_v1", [True, False])
+def test_load_module_module_not_found(snippetcompiler_clean, allow_v1: bool):
+    """
+    Assert behavior when a module is not found as a V1 or a V2 module.
+    """
+    module_name = "non_existing_module"
+    snippetcompiler_clean.modules_dir = None
+    project: Project = snippetcompiler_clean.setup_for_snippet(snippet=f"import {module_name}")
+    with pytest.raises(ModuleNotFoundException, match=f"Could not find module {module_name}"):
+        project.load_module(module_name=module_name, install=True, allow_v1=allow_v1)
+
+
+def test_load_module_v1_and_v2_installed(
+    tmpdir: py.path.local,
+    snippetcompiler_clean,
+    modules_dir: str,
+    caplog,
+) -> None:
+    """
+    Test whether the Project.load_module() method works correctly when the v1 and v2 version of a module are both installed.
+    The V2 module should be loaded and a warning should be logged.
+    """
+    module_name = "minimalv1module"
+    module_dir = os.path.join(modules_dir, module_name)
+
+    # Convert v1 to v2 module
+    module = ModuleV1(project=DummyProject(autostd=False), path=module_dir)
+    module_dir_v2 = os.path.join(tmpdir, f"{module_name}-v2")
+    ModuleConverter(module).convert(output_directory=module_dir_v2)
+
+    project: Project = snippetcompiler_clean.setup_for_snippet(
+        snippet=f"import {module_name}",
+        install_v2_modules=[LocalPackagePath(module_dir_v2, editable=False)],
```

Comment:
How does this work? It doesn't seem to install the v1 module yet the tests are succeeding. What am I missing?

Reply:
The `snippetcompiler` fixture automatically adds the `modules_dir` to the modulepath of the project. So the V1 module is installed implicitly. I will add a comment to clarify this.

Reply:
I see, thanks!